src/org/isf/utils/jobjects/CursorManager.java

```diff
@@ -1,59 +1,32 @@
 package org.isf.utils.jobjects;
 
-import java.awt.*;
+import java.awt.AWTEvent;
+import java.awt.EventQueue;
+import java.awt.Toolkit;
 import java.awt.event.InputEvent;
-import java.lang.management.ManagementFactory;
-import java.util.*;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
-
-import javax.management.MBeanServer;
-import javax.management.MBeanServerBuilder;
-import javax.management.MBeanServerFactory;
-import javax.management.ObjectName;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Stack;
 
 class CursorManager {
-  static EQPermit eqPermit = null;
-  static {
-    try {
-      MBeanServer mbs = ManagementFactory.getPlatformMBeanServer(); 
-      ObjectName name = new ObjectName("org.isf.utils.jobjects:type=EQPermit"); 
-      eqPermit = new EQPermit(); 
-      mbs.registerMBean(eqPermit, name); 
-    }catch(Exception exp){exp.printStackTrace();} 
-    finally {}
-  }
   private final DelayTimer waitTimer;
   private final Stack<DispatchedEvent> dispatchedEvents;
   private boolean needsCleanup;
 
-  private EventQueue parentQueue = null;
-
-  //private Lock clearLock = new ReentrantLock();
-
   public CursorManager(DelayTimer waitTimer) {
     this.dispatchedEvents = new Stack<DispatchedEvent>();
     this.waitTimer = waitTimer;
-    this.parentQueue = Toolkit.getDefaultToolkit().getSystemEventQueue();
   }
   private void cleanUp() {
     if (((DispatchedEvent) dispatchedEvents.peek()).resetCursor()) {
       clearQueueOfInputEvents();
     }
   }
-  private void acquirePermit() throws Exception{
-    if(!eqPermit.getPermit()) 
-          while(!eqPermit.getPermit()) 
-          { 
-            System.out.println("Cant get permit " + eqPermit.getPermit()); 
-            Thread.sleep(2000);
-          }
-  }
   private void clearQueueOfInputEvents() {
-    EventQueue q = Toolkit.getDefaultToolkit().getSystemEventQueue();
+    EventQueue q = Toolkit.getDefaultToolkit().getSystemEventQueue(); 
+    EventQueue parentQueue = ((WaitCursorEventQueue)q).getParentQueue();
     try{
         synchronized(parentQueue){
-          acquirePermit();
           synchronized(q) {
```

Comment:
minor change:
Make it more explicit, so that it is clearer when reading the code:
```
  final WaitCursorEventQueue waitCursorEventQueue = (WaitCursorEventQueue) Toolkit.getDefaultToolkit().getSystemEventQueue();
    final EventQueue parentQueue = waitCursorEventQueue.getParentQueue();
    synchronized(parentQueue){
      synchronized(waitCursorEventQueue) {
      ...
      ..
      }
   }
}
```
(cast to WaitCursorEventQueue earlier, and rename q -> waitCursorEventQueue)

Reply:
Change done. Thanks.